### PR TITLE
Huge update

### PR DESCRIPTION
**Pull request by matteodepalo**

### CODE REVIEW

**Comment by willrax** [comment]
:+1:

**Comment by matteodepalo** [comment]
LOL trying to get some points? :dancer: This PR would break production if deployed :/

**Comment by carlosepp** [comment]
:(

File: app/services/store.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import DS from 'ember-data';
```
Comment (willrax) [inline app/services/store.js]: you can delete this file.

**Comment by willrax** [comment]
@delkopiso looks good. We could try to get rid of the views as they are quite low level and slow.

**Comment by delkopiso** [comment]
@willrax agreed I saw deprecation warnings for it as well.

File: app/router.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import Ember from 'ember';
+import config from './config/environment';
+
+var Router = Ember.Router.extend({
+  location: config.locationType
+});
+
+Router.map(function() {
+  this.route('authenticated', { path: '' }, function() {
+    this.resource('index', { path: '/' });
```
Comment (willrax) [inline app/router.js]: These can all be routes. Also index exists without declaring it when ever you nest a route.

Comment (willrax) [inline app/router.js]: Resource is being deprecated.

Comment (delkopiso) [inline app/router.js]: So something like this:
```javascript
  this.route('authenticated', { path: '' }, function() {
    this.route('leaderboard', { path: '/leaderboard/:time_span' });
    this.route('hall-of-fame');
    this.route('stats');
  });
```

**Comment by matteodepalo** [comment]
:open_mouth:

**Comment by delkopiso** [comment]
Hehe

File: app/router.js (hunk 2 above)
Comment (willrax) [inline app/router.js]: Yep. The path for authenticated would be changed to '/'.

**Pull request merged**